core-plugins/src/main/java/io/cdap/plugin/batch/sink/FileSink.java

```diff
@@ -97,7 +107,12 @@ private Map<String, String> getFSProperties() {
       try {
         return GSON.fromJson(fileSystemProperties, MAP_TYPE);
       } catch (JsonSyntaxException e) {
-        throw new IllegalArgumentException("Unable to parse filesystem properties: " + e.getMessage(), e);
+        String errorMessage = String.format(
+            "Failed to parse filesystem properties %s with message: %s", fileSystemProperties,
```

Comment:
can we please follow the same format everywhere? `%s : %s", e.getClass().getName(), e.getMessage()`